public/entity2/entitysystem.h

```diff
@@ -0,0 +1,246 @@
+	virtual void		BuildResourceManifest(SpawnGroupHandle_t hSpawnGroup, const CUtlVector<const CEntityKeyValues*, CUtlMemory<const CEntityKeyValues*, int> >* pEntityKeyValues, const char* pFilterName, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest) override;
+	virtual void		BuildResourceManifest(SpawnGroupHandle_t hSpawnGroup, int nEntityKeyValueCount, const CEntityKeyValues** ppEntityKeyValues, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest) override;
+	virtual void		UnknownFunc004() override;
+	virtual void		BuildResourceManifestForEntity(uint64 unknown1, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest, uint64 unknown2) override;
+	virtual void		InvokePrecacheCallback(void* hResource, const EntitySpawnInfo_t* const info, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest, char* unk, void* callback) override;
+	virtual void		AddRefKeyValues(const CEntityKeyValues* pKeyValues) override;
+	virtual void		ReleaseKeyValues(const CEntityKeyValues* pKeyValues) override;
+	virtual void		LockResourceManifest(bool bLock, CEntityResourceManifestLock* const context) override;
+	virtual				~CEntitySystem();
+	virtual void		ClearEntityDatabase(ClearEntityDatabaseMode_t eMode);
+	virtual void		FindEntityProcedural(const char* szName, CEntityInstance* pSearchingEntity, CEntityInstance* pActivator, CEntityInstance* pCaller);
+	virtual void		OnEntityParentChanged(CEntityInstance* pEntity, CEntityInstance* pNewParent); // empty function
+	virtual void		OnAddEntity(CEntityInstance* pEnt, CEntityHandle handle); // empty function
+	virtual void		OnRemoveEntity(CEntityInstance* pEnt, CEntityHandle handle); // empty function
+	virtual int			GetSpawnGroupWorldId(SpawnGroupHandle_t hSpawnGroup); // returns 0
+	virtual void		Spawn(int nCount, const EntitySpawnInfo_t* pInfo);
+	virtual void		Activate(int nCount, const EntityActivation_t* pActivates, ActivateType_t activateType);
+	virtual void		PostDataUpdate(int nCount, const PostDataUpdateInfo_t *pInfo);
+	virtual void		OnSetDormant(int nCount, const EntityDormancyChange_t* pInfo, bool bNotifyAddedToPVS);
+	virtual void		UpdateOnRemove(int nCount, const EntityDeletion_t *pDeletion);
+
+public:
+	CBaseEntity* GetBaseEntity(int32_t entnum) // CEntityIndex
+	{
+		if (entnum <= -1 || entnum >= (MAX_TOTAL_ENTITIES - 1))
+			return nullptr;
+
+		CEntityIdentities* pChunkToUse = m_EntityList.m_pIdentityChunks[entnum / MAX_ENTITIES_IN_LIST];
+		if (!pChunkToUse)
+			return nullptr;
+
+		CEntityIdentity* pIdentity = &pChunkToUse->m_pIdentities[entnum % MAX_ENTITIES_IN_LIST];
+		if (!pIdentity)
+			return nullptr;
+
+		if (pIdentity->m_EHandle.GetEntryIndex() != entnum)
+			return nullptr;
+
+		return static_cast<CBaseEntity*>(pIdentity->m_pInstance);
+	}
```

Comment:
I don't think these functions should be defined like that in the header file, it's better to get them their own .cpp file